parsers/ENTSOE.py

```diff
@@ -622,6 +717,35 @@ def fetch_production(zone_key, session=None, target_datetime=None, logger=loggin
     return list(filter(validate_production, data))
 
 
+def fetch_production_per_units(zone_key, session=None, target_datetime=None, logger=logging.getLogger(__name__)):
+    """
+    Returns a list of all production units and production values as a list
+    of dictionaries
+    """
+    if not session:
+        session = requests.session()
+    domain = ENTSOE_EIC_MAPPING[zone_key]
+    data = []
+    # Iterate over all psr types
+    for k in ENTSOE_PARAMETER_DESC.keys():
+        try:
+            values = [ v for v in parse_production_per_units(
+                query_production_per_units(k, domain, session, target_datetime)) or [] if v is not None ]
+            for v in values:
+                if not v: continue
```

Comment:
You're doing two loops in a row over values (counting the list comprehension), twice check if `v` is truthy. One could be enough - I'd remove the list comprehension?
Small detail, PEP 8 recommends avoiding one-liner `if`s.

Reply:
does you linter not take PEP 8 recommandations into account ?

Reply:
Very true. My linter is pretty crappy. I should take some time to upgrade it but basically I'm on my own..